tests/src/Unit/ServerServiceTest.php

```diff
@@ -0,0 +1,148 @@
+ * WSDL which the server service class can be generated from.
+ */
+class ServerServiceTest extends CodeGenerationTestCase
+{
+
+    protected $namespace = 'SoapClientTest';
+
+    // Use our mock soap client. It allows to retrieve the configuration it was passed.
+    protected $soapclientClass = '\Wsdl2PhpGenerator\Tests\Mock\SoapClient';
+
+    // Example Wsdl path.
+    protected $wsdl = '/tmp/some.wsdl';
+
+    // Example options which can be passed as options to a \SoapClient instance.
+    protected $soapclientOptions = array(
+        'features' => SOAP_SINGLE_ELEMENT_ARRAYS,
+        'cache_wsdl' => WSDL_CACHE_NONE,
+    );
+
+    // Use our mock soap server. It allows to retrieve the configuration it was passed.
+    protected $soapserverClass = '\Wsdl2PhpGenerator\Tests\Mock\SoapServer';
+
+    // Example options which can be passed as options to a \SoapServer instance.
+    protected $soapserverOptions = array(
+        'features' => SOAP_SINGLE_ELEMENT_ARRAYS,
+        'cache_wsdl' => WSDL_CACHE_NONE,
+    );
+
+    /**
+     * Test handling of SoapServer configuration.
+     */
+    public function testSoapConfig()
+    {
+        $clientConfig = new Config(array(
+                'inputFile' => $this->wsdl,
+                'outputDir' => null,
+                'namespaceName' => $this->namespace,
+                'soapClientClass' => $this->soapclientClass,
+                'soapClientOptions' => $this->soapclientOptions,
+            ));
+
+        $clientService = new Service($clientConfig, 'TestClientService', array(), 'Service description');
+        $this->generateClass($clientService, $this->namespace);
+
+        $this->assertClassExists('TestClientService', $this->namespace);
+
+        $serverConfig = new Config(array(
+                'inputFile' => $this->wsdl,
+                'outputDir' => null,
+                'namespaceName' => $this->namespace,
+                'serverClassName' => 'TestAbstractServerService',
+                'soapServerClass' => $this->soapserverClass,
+                'soapServerOptions' => $this->soapserverOptions,
+            ));
+
+        $serverService = new ServerService($serverConfig, $clientService->getClass()->getIdentifier(), array(), 'Service description');
+        $this->generateServerServiceClass($serverService, $this->namespace);
+        // PHP-VCR wants us to define the class VCR\Util\SoapClientTest\TestAbstractServerService
+        // before creating \SoapClientTest\TestAbstractServerService.
+        $this->generateServerServiceClass($serverService, "VCR\\Util\\{$this->namespace}", "\\{$this->namespace}\\TestClientService as TestClientService");
+
+        $this->assertClassExists('TestAbstractServerService', $this->namespace);
+
+        eval("namespace {$this->namespace}; class TestServerService extends \\{$this->namespace}\\TestAbstractServerService {}");
```

Comment:
How about using [our code generation library](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/master/lib/PhpSource/PhpClass.php) to generate this instead of string manipulation?